### PR TITLE
HBASE-20060 Add details of off heap memstore into book.

### PR DESCRIPTION
**Pull request by openinx**
The document is just a copy from https://issues.apache.org/jira/browse/HBASE-20460,  Will check all those config keys based on our HBase2.x & master. Thanks.

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 22 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| 0 | mvndep | 33 | Maven dependency ordering for branch |
| +1 | mvninstall | 292 | master passed |
| 0 | refguide | 541 | branch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect. |
| +1 | javadoc | 214 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 13 | Maven dependency ordering for patch |
| +1 | mvninstall | 260 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 1 | The patch has no ill-formed XML file. |
| 0 | refguide | 572 | patch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect. |
| +1 | javadoc | 182 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 11970 | root in the patch failed. |
| +1 | asflicense | 60 | The patch does not generate ASF License warnings. |
| | | 14231 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.snapshot.TestExportSnapshotNoCluster |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-334/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/334 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  refguide  xml  |
| uname | Linux 5a5b6699b6f1 4.4.0-143-generic #169~14.04.2-Ubuntu SMP Wed Feb 13 15:00:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 15ac781057 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| refguide | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-334/1/artifact/out/branch-site/book.html |
| refguide | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-334/1/artifact/out/patch-site/book.html |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-334/1/artifact/out/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-334/1/testReport/ |
| Max. process+thread count | 4925 (vs. ulimit of 10000) |
| modules | C: hbase-common . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-334/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 24 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| 0 | mvndep | 27 | Maven dependency ordering for branch |
| +1 | mvninstall | 250 | master passed |
| +1 | compile | 154 | master passed |
| +1 | checkstyle | 126 | master passed |
| 0 | refguide | 439 | branch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect. |
| +1 | shadedjars | 259 | branch has no errors when building our shaded downstream artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: . |
| +1 | findbugs | 242 | master passed |
| +1 | javadoc | 204 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 12 | Maven dependency ordering for patch |
| +1 | mvninstall | 226 | the patch passed |
| +1 | compile | 157 | the patch passed |
| +1 | javac | 157 | the patch passed |
| +1 | checkstyle | 122 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 1 | The patch has no ill-formed XML file. |
| 0 | refguide | 430 | patch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect. |
| +1 | shadedjars | 258 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 764 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: . |
| +1 | findbugs | 283 | the patch passed |
| +1 | javadoc | 223 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 9502 | root in the patch failed. |
| +1 | asflicense | 69 | The patch does not generate ASF License warnings. |
| | | 14125 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-334/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/334 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  refguide  xml  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 8810ddbbc23e 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / b42614295e |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| refguide | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-334/2/artifact/out/branch-site/book.html |
| findbugs | v3.1.11 |
| refguide | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-334/2/artifact/out/patch-site/book.html |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-334/2/artifact/out/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-334/2/testReport/ |
| Max. process+thread count | 4875 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-server . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-334/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hbase-common/src/main/resources/hbase-default.xml [hunk 1]
```diff
@@ -696,7 +696,30 @@ possible configurations would overwhelm and obscure the important.
...
       a feature which works to prevent heap fragmentation under
       heavy write loads. This can reduce the frequency of stop-the-world
-      GC pauses on large heaps.</description>
+      GC pauses on large heaps.
+    </description>
+  </property>
+  <property>
+    <name>hbase.hregion.memstore.mslab.chunksize</name>
+    <value>2097152</value>
+    <description>The maximum size of a chunk in the MemStoreLAB.</description>
```
Comment (saintstack) [inline hbase-common/src/main/resources/hbase-default.xml]: This is bytes?

File: hbase-common/src/main/resources/hbase-default.xml [hunk 2]
```diff
@@ -696,7 +696,30 @@ possible configurations would overwhelm and obscure the important.
...
+  </property>
+  <property>
+    <name>hbase.regionserver.offheap.global.memstore.size</name>
+    <value>0</value>
```
Comment (saintstack) [inline hbase-common/src/main/resources/hbase-default.xml]: This is megabytes?

File: hbase-common/src/main/resources/hbase-default.xml (hunk 1 above)
Comment (openinx) [reply hbase-common/src/main/resources/hbase-default.xml]: Yeah,  Let me address this in the description. Thanks.

File: hbase-common/src/main/resources/hbase-default.xml (hunk 2 above)
Comment (openinx) [reply hbase-common/src/main/resources/hbase-default.xml]: Yeah,  will also the unit  in doc.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 52 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| 0 | mvndep | 30 | Maven dependency ordering for branch |
| +1 | mvninstall | 249 | master passed |
| +1 | compile | 161 | master passed |
| +1 | checkstyle | 130 | master passed |
| 0 | refguide | 445 | branch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect. |
| +1 | shadedjars | 263 | branch has no errors when building our shaded downstream artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: . |
| +1 | findbugs | 252 | master passed |
| +1 | javadoc | 197 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 11 | Maven dependency ordering for patch |
| +1 | mvninstall | 242 | the patch passed |
| +1 | compile | 174 | the patch passed |
| +1 | javac | 174 | the patch passed |
| +1 | checkstyle | 135 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 2 | The patch has no ill-formed XML file. |
| 0 | refguide | 460 | patch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect. |
| +1 | shadedjars | 275 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 791 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: . |
| +1 | findbugs | 307 | the patch passed |
| +1 | javadoc | 236 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 16009 | root in the patch failed. |
| +1 | asflicense | 65 | The patch does not generate ASF License warnings. |
| | | 20839 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.master.assignment.TestOpenRegionProcedureHang |
|   | hadoop.hbase.client.TestFromClientSide |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-334/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/334 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  refguide  xml  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 47cac8c959da 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / b42614295e |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| refguide | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-334/4/artifact/out/branch-site/book.html |
| findbugs | v3.1.11 |
| refguide | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-334/4/artifact/out/patch-site/book.html |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-334/4/artifact/out/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-334/4/testReport/ |
| Max. process+thread count | 4698 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-server . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-334/4/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request merged**